hazelcast-jet-core/src/main/java/com/hazelcast/jet/json/JsonUtil.java

```diff
@@ -62,129 +75,135 @@ private JsonUtil() {
+     * Returns a function which takes a file {@code Path} as input and
+     * returns a stream of objects with the given type. The content of the file
+     * is considered to have a sequence of JSON strings, each one can span
+     * multiple lines. The function is designed to be used with
+     * {@link FileSourceBuilder#build(FunctionEx)}.
+     * <p>
+     * See {@link Sources#json(String, Class)}.
      */
     @Nonnull
-    public static List<Object> getList(@Nonnull String jsonString, @Nonnull String key) {
-        return (List) parse(jsonString).get(key);
+    public static <T> FunctionEx<? super Path, ? extends Stream<T>> asMultilineJson(
+            @Nonnull Class<T> type
+    ) {
+        return path -> {
+            InputStreamReader reader = new InputStreamReader(new FileInputStream(path.toFile()), UTF_8);
+            Spliterator<T> spliterator = Spliterators.spliteratorUnknownSize(JsonUtil.sequenceFrom(type, reader),
+                    Spliterator.ORDERED | Spliterator.NONNULL);
+            return StreamSupport.stream(spliterator, false);
+        };
     }
 
     /**
-     * Extracts an array value as a {@link Object Object[]} from given JSON
-     * string. For extracting multiple values from a JSON string see
-     * {@link #parse(String)}.
+     * Returns a bi-function which takes the fileName and line as input
+     * and returns an object by converting the line to an object of
+     * given {@code type}. The function is designed to be used with
+     * {@link FileSourceBuilder#build(BiFunctionEx)} and
+     * {@link FileSourceBuilder#buildWatcher(BiFunctionEx)}.
+     * <p>
+     * See {@link Sources#json(String, Class)} and
+     * {@link Sources#jsonWatcher(String, Class)}.
      */
     @Nonnull
-    public static Object[] getArray(@Nonnull String jsonString, @Nonnull String key) {
-        return getList(jsonString, key).toArray();
+    public static <T> BiFunctionEx<String, String, ? extends T> asJson(@Nonnull Class<T> type) {
+        return (fileName, line) -> JsonUtil.mapFrom(type, line);
     }
 
     /**
-     * Extracts an object as a {@link Map} from given JSON string. For
-     * extracting multiple values from a JSON string see {@link #parse(String)}.
+     * Returns a bi-function which takes the fileName and line as input, parses
+     * the line to the {@link Map} representation of the JSON string and
+     * returns it. The function is designed to be used with
+     * {@link FileSourceBuilder#build(BiFunctionEx)} and
+     * {@link FileSourceBuilder#buildWatcher(BiFunctionEx)}.
+     * <p>
+     * See {@link Sources#json(String)} and
+     * {@link Sources#jsonWatcher(String)}.
      */
     @Nonnull
-    public static Map<String, Object> getObject(@Nonnull String jsonString, @Nonnull String key) {
-        return (Map<String, Object>) parse(jsonString).get(key);
+    public static BiFunctionEx<String, String, Map<String, Object>> asJson() {
```

Comment:
I think we can remove this function, since it doesn't add much, filename is just ignored?